saml/gatein-saml-plugin/src/main/java/org/gatein/sso/saml/plugin/valve/AbstractSPFormAuthenticator.java

```diff
@@ -692,6 +707,31 @@ private boolean generalUserRequest(Request request, Response response, LoginConf
 
         return localAuthentication(request, response, loginConfig);
     }
+    
+    /**
+     * Extract Gatein roles to put in Principal
+     * @param userId
+     * @return
+     */
+    private List<String> extractGateinRoles(String userId) {
+        OrganizationService organizationService =
+            PortalContainer.getInstance().getComponentInstanceOfType(OrganizationService.class);
+        RolesExtractor rolesExtractor=PortalContainer.getInstance().getComponentInstanceOfType(RolesExtractor.class);
+        List<String> result = new ArrayList<>();
+        Set<MembershipEntry> entries = new MembershipHashSet();
+        Collection<Membership> memberships;
+        try {
+            memberships = organizationService.getMembershipHandler().findMembershipsByUser(userId);
+        } catch (Exception e) {
```

Comment:
IMO, we can add the exception in DEBUG level to help diagnose issues if they happen.

Reply:
Updated